docs/development/CodingStyle.md

```diff
@@ -0,0 +1,253 @@
+One blank line should follow the declaration(s).
+
+Variables with limited use may be declared at the point of first use. It makes PR-review easier (but that point is lost if the variable is used everywhere anyway).
+
+##Initialisation
+The pattern with "lazy initialisation" may be advantageous in the Configurator to speed up the start when the initialisation is "expensive" in some way.
+In the FC, however, it’s always better to use some milliseconds extra before take-off than to use them while flying.
+
+So don't use "lazy initialisation".
+
+An explicit "init" function, is preferred.
+
+##Data types
+Be aware of the data types you use and do not trust implicit type casting to be correct.
+
+Angles are sometimes represented as degrees in a float. Sometimes as decidegrees in a uint8_t.
+You have been warned.
+
+Avoid implicit double conversions and only use float-argument functions.
+
+Check .map file to make sure no conversions sneak in, and use -Wdouble-promotion warning for the compiler
+
+Instead of sin() and cos(), use sin_approx() and cos_approx() from common/math.h.
+
+Float constants should be defined with "f" suffix, like 1.0f and 3.1415926f, otherwise double conversion might occur.
+
+#Functions
+
+##Naming
+Methods that return a boolean should be named as a question, and should not change any state. e.g. 'isOkToArm()'.
+
+Methods should have verb or verb-phrase names, like deletePage or save. Tell the system to 'do' something 'with' something. e.g. deleteAllPages(pageList).
+
+Non-static functions should be prefixed by their class. Eg baroUpdate and not updateCompass .
+
+Groups of functions acting on an 'object' should share the same prefix, e.g.
+```
+float biQuadFilterApply(...);
+void biQuadFilterInit(...);
+boolean biQuadIsReady();
+```
+rather than
+```
+float applyBiQuadFilter(...);
+void newBiQuadLpf(...);
+boolean isBiQuadReady();
+```
+
+##Parameter order
+Data should move from right to left, as in memcpy(void *dst, const void *src, size_t size).
+This also mimics the assignment operator (e.g. dst = src;)
+
+When a group of functions act on an 'object' then that object should be the first parameter for all the functions, e.g.:
+```
+float biQuadFilterApply(biquad_t *state, float sample);
+void biQuadNewLpf(biquad_t *state, float filterCutFreq, uint32_t refreshRate);
+```
+rather than
+```
+float biQuadFilterApply(float sample, biquad_t *state);
+void biQuadNewLpf(float filterCutFreq, biquad_t *state, uint32_t refreshRate);
+```
+
+##Declarations
+Functions not used outside their containing .c file should be declared static (or STATIC_UNIT_TESTED so they can be used in unit tests).
+
+Non-static functions should have their declaration in a single .h file.
+
+Don't make more than necessary visible for other modules, not even types. Pre-processor macros may be used to declare module internal things that must be shared with the modules test code but otherwise hidden.
+
+In the .h file:
+```
+#ifdef MODULENAME_INTERNALS_
```

Comment:
I did use `<module>_impl.h ` for this, it does hide the internals a bit more. But it is easy to convert it to this format

Also some things need to be declared due to C syntax, bus should be hidden from user - how to handle that ?